lib/WTSI/NPG/Data/BamDeletion.pm

```diff
@@ -0,0 +1,325 @@
+);
+
+=head2 md5_file
+
+=cut
+
+has 'md5_file' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 0,
+  lazy_build =>1
+);
+sub _build_md5_file {
+    my $self = shift;
+    my $filename = fileparse($self->file);
+    my $md5file  = $self->outdir.qq[/$filename].q[.md5];
+return $md5file;
+}
+
+sub _build_md5sum {
+    my ($self) = @_;
+    my $md5 = Digest::MD5->new;
+    my $fh;
+    open $fh, '<', $self->outfile ||
+        $self->logcroak(q[Failed to open path '], $self->outfile, q[']);
+        $md5->addfile($fh);
+    close $fh ||
+        $self->logcroak(q[Failed to close path '], $self->outfile, q[']);
+
+    my $md5_fh;
+    open $md5_fh, '>', $self->md5_file ||
+         $self->logcroak(q[Failed to open md5 file '], $self->md5_file, q[']);
+         $self->logwarn(q[MD5 file ],$self->md5_file);
+    print $md5_fh $md5->hexdigest || $self->logcroak(q[Failed to print to md5 file '], $self->md5_file, q[']);;
+    close $md5_fh ||
+          $self->logcroak(q[Failed to close md5 file '], $self->md5_file, q[']);
+    return $md5->hexdigest;
+}
+
+
+=head2 outdir
+
+Where to write data stripped file prior to loading to iRODS
+
+=cut
+
+has 'outdir' => (
+    isa        => 'Str',
+    is         => 'ro',
+    required   => 0,
+    default    => tempdir( CLEANUP => 1 ),
+    documentation => 'Output directory for header/comment file. Default is a directory in /tmp'
+);
+
+=head2 outfile
+
+Locally written file
+
+=cut
+
+has 'outfile' => (
+         isa        => 'Str',
+         is         => 'ro',
+         required   => 0,
+         lazy_build => 1,
+);
+sub _build_outfile {
+    my $self = shift;
+    my $filename = fileparse($self->file);
+    my $outfile     = $self->outdir.qq[/$filename];
+return $outfile;
+}
+
+
+
+=head2 process
+
+Processes files for samples where consent has been withdrawn.
+
+=cut
+
+
+
+sub process {
+  my $self = shift;
+
+  $self->dry_run and $self->info('DRY RUN - no data removal');
+
+    ###check file suffix
+   if ($self->file =~ /[b|cr]am$/sxm){
+      $self->_write_header($self->_generate_header());
+    } else {
+      $self->_write_stub_file();
+    }
+     $self->_write_md5_file();
+
+     if ($self->dry_run){ carp q[Would be re-loading file ], $self->file,qq[\n] }
+     else { $self->_reload_file(); }
+
+  return $self->file;
+}
+
+sub _generate_header{
+    my $self = shift;
+
+    my $obj = WTSI::NPG::HTS::Illumina::AlnDataObject->new($self->irods,$self->file);# sub header
+
+       $obj->is_present or $self->info($self->file ,q[ not found]);
+
+    my $header = $obj->header;
+return $header;
+}
+
+
+sub _write_header{
+    my $self = shift;
+    my $header = shift;
+    push @{$header}, $DELETION_MSG;
+    if ($self->dry_run){ carp ("Would be writing header and \"$DELETION_MSG\" to ",$self->outfile) ; return }
+    write_file($self->outfile,map { "$_\n" } @{$header}) or $self->logcroak(q[Cannot write ] ,$self->outfile);
+return 1;
```

Comment:
Indent?
The return value is being used only in the tests. That affects the clarity of the code and also makes the test
fragile. If someone removes the return because it isn't used, the test will fail even if the function is correct.
The test is also giving a false sense of security because it doesn't actually test that anything is written (there's
a different test doing that).